Test with real git dir

rjmunro opened the pull request:
New code fails with fake .git dir as it doesn't guess, it asks git.

tarmolov commented on tests/install.test.js:
```diff
@@ -1,15 +1,17 @@
 var fsHelpers = require('../lib/fs-helpers');
 
-var SANDBOX_PATH = __dirname + '/tmp-sandbox/';
+var SANDBOX_PATH = '/tmp/tmp-sandbox/';
```
Let's use `mktemp -d` here. It creates a temporary directory and returs its name.

rjmunro commented:
@tarmolov I've fixed the uninstall test and added a new test for a missing hooks folder, so I think the coverage is as high as it can be. I can't figure out a way to cause an exception to cover the last uncovered line. I'm tempted to just remove the try ... catch to get the 100% coverage, but I don't think that would be good practise :-).

It may be possible to get the 100% coverage by converting to promises instead of callbacks, as they could handle the exception automatically, but that would require quite a large rework.

tarmolov commented:
Thank you!
